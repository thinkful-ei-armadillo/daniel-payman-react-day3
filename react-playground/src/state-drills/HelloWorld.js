import React from 'react';

class HelloWorld extends React.Component {
  state = {
    who: 'world'
  };

  handleClickWorld = () => {
    this.setState({ who: 'world' });
  };

  handleClickFriend = () => {
    this.setState({ who: 'friend' });
  };

  handleClickReact = () => {
    this.setState({ who: 'react' });
  };

  render() {
    return (
      <div className="helloWorld">
        <button onClick={this.handleClickWorld} className="world">
          World
        </button>
        <button onClick={this.handleClickFriend} className="friend">
          Friend
        </button>
        <button onClick={this.handleClickReact} className="react">
          React
        </button>

        <p>Hello, {this.state.who}!</p>
      </div>
    );
  }
}

export default HelloWorld;
