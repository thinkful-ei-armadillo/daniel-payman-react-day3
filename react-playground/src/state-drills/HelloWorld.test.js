import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

describe('HelloWorld component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HelloWorld />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<HelloWorld name="helloWorld"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});