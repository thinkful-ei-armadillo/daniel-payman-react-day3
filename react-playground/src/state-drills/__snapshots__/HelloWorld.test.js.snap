// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld component renders the UI as expected 1`] = `
<div
  className="helloWorld"
>
  <button
    className="world"
    onClick={[Function]}
  >
    World
  </button>
  <button
    className="friend"
    onClick={[Function]}
  >
    Friend
  </button>
  <button
    className="react"
    onClick={[Function]}
  >
    React
  </button>
  <p>
    Hello, 
    world
    !
  </p>
</div>
`;
